cromwell-drs-localizer/src/main/scala/drs/localizer/downloaders/AccessUrlDownloader.scala

```diff
@@ -0,0 +1,28 @@
+package drs.localizer.downloaders
+
+import cats.effect.{ExitCode, IO}
+import cloud.nio.impl.drs.AccessUrl
+import com.typesafe.scalalogging.StrictLogging
+
+import scala.sys.process.{Process, ProcessLogger}
+
+case class AccessUrlDownloader(accessUrl: AccessUrl, downloadLoc: String) extends Downloader with StrictLogging {
+
+  def generateDownloadScript(): String = {
+    val signedUrl = accessUrl.url
+    // TODO headers, requester pays, refinements to retry strategy for certain 4xx statuses
+    s"""mkdir -p $$(dirname '$downloadLoc') && curl --location --retry 3 --retry-connrefused --retry-delay 10 --fail --output '$downloadLoc' '$signedUrl'"""
+  }
+
+  override def download: IO[ExitCode] = {
+    val signedUrl = accessUrl.url
+      // TODO probably don't want to log the actual signed URL
+      logger.info(s"Attempting to download $signedUrl to $downloadLoc")
```

Comment:
When is this TODO going to be addressed? Before the PR merges or does it need a future Jira ticket?

ToL: One can discuss with appsec as they might have recommendations of the safety of options like:
- log the URL anyway since it expires, or...
- log the URL without the query string

Reply:
We touched on this during tech-talk this morning. Since this is in the localizer, I'm not sure where these logs are going... maybe an object in the workspace bucket, maybe Google Cloud Logging... either way, it probably doesn't matter. Access to these objects are tied to an individual's connection to the data, which is even more specific than workspace access. Therefore, I think we should not log the full signed URL.

I think we should still log _something_ to aid debugging of localization. Logging the DRS URI and the decision to download via signed URL might be enough... or maybe chopping off any query string from the signed URL would make it safe enough to log.


... though I just realized that this is the _localizer_ we're talking about. Where do these logs go?

Reply:
I believe this output should make it to the `task_name.log` file that winds up under the user's "folder" for the associated task in GCS ([example log](https://00f74ba44b93a7a7fc4c2de08ea29c0b680795f665-apidata.googleusercontent.com/download/storage/v1/b/cloud-cromwell-dev-self-cleaning/o/cromwell_execution%2Fci%2Fcall_cache_cha_cha%2F06e44cfb-8d6f-47ab-a9b0-57f902eb0ae3%2Fcall-read_files_new_command%2Fread_files_new_command.log?jk=AFshE3W5xy0F1pE5Dfo2_7MuIr4UTVn5KmtNhjrimzxI47kL7hDvdV7FRYCN7WvfLnTvxjblyyXEW5l-GyvwPT27UvxKrD-z1poemc-UlncMM4wAh17iPrWY6kg_IH3m5oVjkhvDFBwWBGp33qdb_PpiyAe_jeJnvnLfe9poX-IXBUgbj2EML_RZ3BEcxxavAb3BwikM8lgYTrmAJEitqed-WyfEkhQfGYU3OQZ2mSpFjkolin4d6Xzue_47lsYP5kY_UilDBAvpKASqMXO0YFakLhXb2so5NsxvejzrtPuTS6bZPPgfz5--e0yZ0MyvDdbaYm-oy1YySUUrgiPRyB6sy8CeQGXq5BKHYi1Tt1uBvIAhVB0z2LL3sAn4cik7UmcP2OHeEQwyGZPjtcYCxtltda4g_yLcSMHQuTZPszVHeATJo19nHWf4rJ0nCAnv8rgdUMLb-E9lnUbMHqK2G33B9Qk3TsEggR5nw21ICBSAYYTl5eu1a63iBSeyenzlyMoRHhftB-RsAWk0Kwi6dE1VNYdwtErRqRYoFq6u5pg7uXiCuXBiB7ZkLrQ131w_5Fo2akwKZmHNWlDoV5NxBaIF-wPWWSyQ9ilOxgz-Z7RrNTPlcK2LtVoEgWT5FE_074xrs_4w9VZ-TXi3kO_GLHkl3GVDl41YRbCR78W-l9EPwpYRwQ_Jdl3qMqmi5iC3iRjmmIcUQjyp75lHxIiu86uzUBEgKV1FQGNBDTOFVxlS0zO0LA7cZXco54n0C6mMmkYmIHns13kZplfHqEbvPMUnWdR8DSkWB7nldlR7WAWIJLXdb7kiu82ShhM2rrxsT9FfoaVfoo8x6wxO42RB-7M-sigSuTNjNadiJ8AhS9_1FsNdWdHHDqK_D3GCo3dFkAO8cP0LMMEQb7iFvDUzxffrgeAcEsKsy-OK8zsqZqkn73n7X39yTLcXSbyn0QdqJBeihrm4B0Hq6ffP-l25hfJmRfbrpT4fgKs2NLTYhTx8hYRTXticsE6yttdBIWYmoTXkJIUFlyYb1eSqHqMtWZjbNXMJvz_XRVXDyOXieW6oG_r2vCsUofmjixEdCHr83pDmVnTqDqWeY3mKrcwpwdhaJIldfAPxBYnd0M3UPmshN3-pYjbT-gbpVA8iHCuEXa-t0Eda79E&isca=1) from a CI run). This should be a very safe place as it will already contain other sensitive data such as the outputs from this task. So I'm kinda wondering if we really need to change this line after all...

Reply:
The logs for the DRS localizers (AccessURL or GCS) go into the same GCS bucket that contains the other call logs. Below are steps to get to the GCS DRS localizer log from the CI for this current PR.

- Start with the [Travis Logs](https://travis-ci.com/github/broadinstitute/cromwell/jobs/499764041)
- Search for `Submitting drs_usa_jdr returned workflow id`
- Get workflow ID = `a2699805-0a16-479f-bacb-f2bfa8755a4b`
- CI uses at output [root](https://github.com/broadinstitute/cromwell/blob/61/src/ci/resources/papi_v1_v2alpha1_provider_config.inc.conf#L2) of `gs://cloud-cromwell-dev-self-cleaning/cromwell_execution/ci`
- So path to this workflow outputs is `gs://cloud-cromwell-dev-self-cleaning/cromwell_execution/ci/drs_usa_jdr/a2699805-0a16-479f-bacb-f2bfa8755a4b`
- Path to aggregated Papi action logs for the call is `gs://cloud-cromwell-dev-self-cleaning/cromwell_execution/ci/drs_usa_jdr/a2699805-0a16-479f-bacb-f2bfa8755a4b/call-skip_localize_jdr_drs_with_usa/skip_localize_jdr_drs_with_usa.log`
- The individual download action just happened to be action 13 (one can iterate through the actions until one finds the individual action logs, or can use the gcloud papi operations info to list the actions) revealing the individual action log is `gs://cloud-cromwell-dev-self-cleaning/cromwell_execution/ci/drs_usa_jdr/a2699805-0a16-479f-bacb-f2bfa8755a4b/call-skip_localize_jdr_drs_with_usa/pipelines-logs/action/13/stdout`